### PR TITLE
When loading data into a form, make sure its using ALL fields

### PR DESCRIPTION
**Pull request by dhensby**
Fixes issue where loading data into a form doesn't take into account fields that could have been added by `getExtraFields`.

### CODE REVIEW

File: tests/forms/FormTest.php [hunk 1]
```diff
@@ -673,41 +693,57 @@ public function doSubmit($data, $form, $request) {
...
-				new FormAction('doSubmit')
-			)
-		);
-		$form->setFormMethod('POST');
-		$form->setStrictFormMethodCheck(true);
-		$form->disableSecurityToken(); // Disable CSRF protection for easier form submission handling
+class FormTest_ExtraFieldsForm extends Form implements TestOnly {
+
+    public function getExtraFields() {
+        $fields = $this->getExtraFields();
```
Comment (chillu) [inline tests/forms/FormTest.php]: @dhensby This causes recursion

Comment (dhensby) [inline tests/forms/FormTest.php]: lol, yes, should be `parent::getExtraFields`

**Comment by chillu** [comment]
Technically, the `TEST` commit should be merged into the `FIX` one, they're dealing with the same thing. Also, there's no `TEST` "commit tag" documented in SilverStripe, and I don't see any value in declaring it like this. But these are both minor niggles, so merging as-is :)

**Comment by dhensby** [comment]
@chillu - I like to do a test commit that fails first to prove that the fix works, I then usually fixup the commits into one

**Pull request merged**